cmd/goProbe/goProbe.go

```diff
@@ -52,11 +52,12 @@ func main() {
 	}
 
 	// Read / parse config file
-	config, err := gpconf.ParseFile(flags.CmdLine.Config)
+	configMonitor, err := gpconf.NewMonitor(flags.CmdLine.Config)
```

Comment:
As you can see, the required changes to the calling path are minimal (and from what I can see there are no side effects from this because the `config` variable is only used for the initial setup. After that, the only interaction / access to the config is via the `Reload()` / callback function of the monitor (hooking into the `CaptureManager`).